### PR TITLE
ENH: Add notebook with MONAI-affine and Elastix-bspline

### PR DESCRIPTION
**Pull request by ntatsisk**
To be updated when:

- https://github.com/Project-MONAI/MONAI/pull/5897 is merged into MONAI (original PR here: https://github.com/InsightSoftwareConsortium/itk-torch-transform-bridge/pull/6)
- Elastix supports `SetInitialTransform(itk_transform)`

Related to #126

### CODE REVIEW

File: examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb [hunk 1]
```diff
@@ -0,0 +1,577 @@
+{
```
Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.    
</pre><p>Maybe add a comment here, saying why set_determinism(42) is called. I guess:</p><p><code>#Set the random seed to a fixed number, just to ensure that the example always produces the same result </code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #25.        max_epochs=3
</pre><p>Maybe add a comment, saying that the number of epochs is limited for CPU runs (and thereby suboptimal), because otherwise running the example would take too much time.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.            LoadImageD(keys=["fixed_hand", "moving_hand"]),
</pre><p>Could this warning be avoided? By adding image_only=False? Warning: <em>"monai.transforms.io.dictionary LoadImaged.__init__:image_only: Current default value of argument </em><code><em>image_only=False</em></code><em> as been deprecated since version 1.1. It will be changed to</em><code><em>image_only=True</em></code><em> in version 1.3."</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.            ScaleIntensityRangeD(keys=["fixed_hand", "moving_hand"],
</pre><p>Dummy beginner question: how do these XxxD functions "know" that they are applied to the variable <code>training_datadict</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #14.                        #scale_range=(-0.1, 0.1),
</pre><p>May this comment be removed, or does it suggest a future experiment or tweak? (Same for mode=nearest)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            super().__init__()
</pre><p>Dummy question, I don't see any superclass, so why doing an init()? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #34.    if USE_COMPILED:
</pre><p>Maybe add a comment here explaining when USE_COMPILED is true, and why it is treated differently </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (N-Dekker) [inline examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Perform numerical comparison
</pre><p>Mention "mean squared error (MSE)" </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/InsightSoftwareConsortium/ITKElastix/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (ntatsisk) [reply examples/ITK_Example18_MONAI_affine_Elastix_nonlinear.ipynb:1]: <p>At this point it doesn't. However, in the next cell we do:</p><p> </p><pre class="ql-syntax" spellcheck="false">dataset = CacheDataset(data=training_datadict, transform=train_transforms)
dataloader = DataLoader(dataset=dataset, batch_size=16, shuffle=True)
</pre><p>where we supply the mentioned variable.</p>

**Review by thewtex — APPROVED** [review]
:sparkles:
